src/topology/local_homeomorph.lean

```diff
@@ -540,7 +540,8 @@ by ext x; simp [prod_coe_symm]
   (e : local_homeomorph α β) (f : local_homeomorph β γ)
   (e' : local_homeomorph δ η) (f' : local_homeomorph η ε) :
   (e.prod e').trans (f.prod f') = (e.trans f).prod (e'.trans f') :=
-by ext x; simp [ext_iff]; tauto
+local_homeomorph.eq_of_local_equiv_eq $
+  by simp only [trans_to_local_equiv, local_equiv.prod_trans, prod_to_local_equiv]
```

Comment:
You can, but even if you give the arguments the direct term proof is slower than rewriting. This is the fastest version I was able to find, around 22ms as opposed to 170ms for `local_homeomorph.eq_of_local_equiv_eq $ local_equiv.prod_trans _ _ _ _`:
```suggestion
  by dsimp only [trans_to_local_equiv, prod_to_local_equiv]; apply local_equiv.prod_trans
```

Reply:
Thanks for finding an even faster proof, Mario!